drain3/template_miner.py

```diff
@@ -170,24 +171,93 @@ def match(self, log_message: str, full_search_strategy="never") -> LogCluster:
         matched_cluster = self.drain.match(masked_content, full_search_strategy)
         return matched_cluster
 
-    def get_parameter_list(self, log_template: str, content: str):
+    def get_parameter_list(self, log_template: str, content: str, exact_matching: bool = True) -> [str]:
+        """
+        Extract parameters from a log message according to a provided template.
+        This will work best if each `MaskingInstruction` has a `mask_with` unique accross all instances
+        and none of them have a value of `*`.
+
+        :param log_template: log template corresponding to the log message
+        :param content: log message to extract parameters from
+        :param exact_matching: whether to apply the correct masking-patterns to match parameters, or try to approximate;
+            turning exact_matching may be faster but may lead to situations in which the parameters are wrongly identified
+        :return: A list of parameters present in the log message.
+        """
+        return [parameter for parameter, mask in self.get_parameter_and_mask_list(log_template, content, exact_matching=exact_matching)]
+
+    def get_parameter_and_mask_list(self, log_template: str, content: str, exact_matching: bool = True) -> [(str, str)]:
+        """
+        Extract parameters and their corresponding mask from a log message according to a provided template.
+        This will work best if each `MaskingInstruction` has a `mask_with` unique accross all instances
+        and none of them have a value of `*`.
+
+        :param log_template: log template corresponding to the log message
+        :param content: log message to extract parameters from
+        :param exact_matching: whether to apply the correct masking-patterns to match parameters, or try to approximate;
+            turning exact_matching may be faster but may lead to situations in which the parameters are wrongly identified
+        :return: A list of parameters present in the log message.
+        """
+        simple_parameter_spec = (r".*?", [])
+        # Gather all masking-patterns defined together with the names of their named regex-groups.
+        mask_patterns = {}
+        for instruction in self.config.masking_instructions:
```

Comment:
the block of building mask_patterns dictionary should be called only one, in the ctor. Thjis should improve performance. Also refactor to a separate method.